test/tests/conditional_tests.py

```diff
@@ -0,0 +1,287 @@
+        test_list.append(test_cfg)
+
+        # Test 4:
+        # Only_if that fails to skip with deferred not_if that skips.
+        test_cfg = base_cfg.copy()
+        test_cfg['only_if'] = {'{{dumb_user}}': ['nivlac', 'calvin'],
+                               '{{dumb_os}}': ['bieber']}
+        test_cfg['not_if'] = {'{{dumb_sys_var}}': ['stupid']}
+        test_list.append(test_cfg)
+
+        # Run through scenario of deferred(no-skip) into skip.
+        for test_cfg in test_list:
+            test = self._quick_test(cfg=test_cfg, finalize=False)
+            self.assertFalse(test.skipped, msg="dumb_sys_var should be deferred"
+                                               " with skip not assigned to"
+                                               " the test")
+
+            fin_sys = system_variables.SysVarDict(defer=False, unique=True)
+            fin_var_man = VariableSetManager()
+            fin_var_man.add_var_set('sys', fin_sys)
+            test.finalize(fin_var_man)
+            self.assertTrue(test.skipped, msg="Now it should skip")
+
+        test_list = []
+        # Test 5:
+        # Not_if with deferred variable that resolves to  no skip.
+        test_cfg = base_cfg.copy()
+        test_cfg['not_if'] = {'{{dumb_sys_var}}': ['notstupid']}
+        test_list.append(test_cfg)
+
+        # Test 6:
+        # Only_if with deferred variable that resolves to no skip.
+        test_cfg = base_cfg.copy()
+        test_cfg['only_if'] = {'{{dumb_sys_var}}': ['stupid']}
+        test_list.append(test_cfg)
+
+        # Test 7:
+        # Not_if and only_if-deferred that fails to skip.
+        test_cfg = base_cfg.copy()
+        test_cfg['not_if'] = {'{{dumb_user}}': ['nivlac', 'notcalvin'],
+                              '{{dumb_os}}': ['blurg']}
+        test_cfg['only_if'] = {'{{dumb_sys_var}}': ['stupid']}
+        test_list.append(test_cfg)
+
+        # Test 8:
+        # Only_if and not_if-deferred that fails to skip.
+        test_cfg = base_cfg.copy()
+        test_cfg['only_if'] = {'{{dumb_user}}': ['nivlac', 'calvin'],
+                               '{{dumb_os}}': ['bieber']}
+        test_cfg['not_if'] = {'{{dumb_sys_var}}': ['notstupid']}
+        test_list.append(test_cfg)
+
+        # Run through scenario of deferred(no-skip) into no skip.
+        for test_cfg in test_list:
+            test = self._quick_test(cfg=test_cfg, finalize=False)
+            self.assertFalse(test.skipped, msg="dumb_sys_var should be deferred"
+                                               " with skip not assigned to"
+                                               " the test.")
+
+            fin_sys = system_variables.SysVarDict(defer=False, unique=True)
+            fin_var_man = VariableSetManager()
+            fin_var_man.add_var_set('sys', fin_sys)
+            test.finalize(fin_var_man)
+            self.assertFalse(test.skipped, msg="Test Should NOT skip.")
+
+    def test_regex(self):
```

Comment:
Just merge this into your tests above.

Reply:
Your other tests ensure that the basic functionality works. All you need to check for here is that the regex's actually work beyond just a simple string.